spec/keycard/request/attributes_spec.rb

```diff
@@ -4,104 +4,113 @@
-    let(:inst_attributes) { { inst_attr: 'value', foo: 'bar' } }
+  describe "#base" do
+    it "includes our default attrs" do
+      expect(attributes.base).to eql({
+        user_pid:  nil,
+        user_eid:  nil,
+        client_ip: nil,
+        auth_token: auth_token
+      })
 
-    it "can get the value of an attribute" do
-      expect(request_attributes[:inst_attr]).to eq('value')
-    end
-
-    it "merges identity attributes according to the finder's identity_keys" do
-      expect(request_attributes.identity[:inst_attr]).to eq 'value'
     end
+  end
 
-    it "passes through supplemental attributes" do
-      expect(request_attributes.supplemental[:foo]).to eq('bar')
+  describe "#[]" do
+    it "can get the value of a finder attribute" do
+      expect(attributes[:finder_attr]).to eql('value')
     end
   end
 
   describe "#all" do
-    let(:user_attributes) { { username: 'user' } }
-    let(:inst_attributes) { { baz: 'quux' } }
+    let(:base_attributes) { { user_pid: 'pid', user_eid: 'eid', client_ip: '10.0.0.1', zilch: nil } }
+    let(:finder_attributes) { { baz: 'quux', nada: nil } }
+    before(:each) { allow(attributes).to receive(:base).and_return(base_attributes) }
 
-    it "returns only the institutional attribute because pid, eid, and IP are nil" do
-      expect(request_attributes.all).to eq inst_attributes
+    it "ignores nil attributes" do
+      expect(attributes.all.keys).to_not include(:nada, :zilch)
     end
-  end
 
-  describe "#identity" do
-    let(:base_attributes) do
-      { user_pid: 'pid', user_eid: 'eid', client_ip: '10.0.0.1' }
+    it "includes base attributes and those from finders" do
+      expect(attributes.all).to eql(base_attributes.merge(baz: 'quux'))
     end
-    let(:inst_attributes) { { baz: 'quux' } }
+  end
 
-    it "returns the identity attributes" do
-      allow(request_attributes).to receive(:base).and_return(base_attributes)
-      expect(request_attributes.identity).to eq(user_pid: 'pid', user_eid: 'eid')
+  describe "#identity" do
+    let(:base_attributes) { { user_pid: 'pid', user_eid: 'eid', client_ip: '10.0.0.1' } }
+    before(:each) { allow(attributes).to receive(:base).and_return(base_attributes) }
+
+    it "includes pairs according to the identity_keys" do
+      expect(attributes.identity).to eql(
+        user_pid: 'pid',
+        user_eid: 'eid',
+        finder_attr: 'value'
+      )
     end
   end
 
   describe "#supplemental" do
     context "with the default supplemental attributes" do
       let(:base_attributes) { { user_pid: 'user', client_ip: '10.0.0.1' } }
-      let(:inst_attributes) { { baz: 'quux' } }
-      let(:attributes)      { { baz: 'quux' } }
+      before(:each) { allow(attributes).to receive(:base).and_return(base_attributes) }
 
-      it "returns the supplemental attributes" do
-        expect(request_attributes.supplemental).to eq attributes
+      it "returns non-identity attributes" do
```

Comment:
Nice catch. Let's use a parenthetical until we can decide the exactly naming later. That is, "supplemental (non-identity)".